apis/r/R/SOMAExperimentAxisQuery.R

```diff
@@ -169,15 +169,210 @@ SOMAExperimentAxisQuery <- R6::R6Class(
       )
     },
-    #' @description ...
+    #' @description Loads the query as a \code{\link[SeuratObject]{Seurat}} object
     #'
-    #' @param X_layers ...
+    #' @param X_layers A named character of X layers to add to the Seurat assay;
+    #' names should be one of:
+    #' \itemize{
+    #'  \item \dQuote{\code{counts}} to add the layer as \code{counts}
+    #'  \item \dQuote{\code{data}} to add the layer as \code{data}
+    #'  \item \dQuote{\code{scale.data}} to add the layer as \code{scale.data}
+    #' }
+    #' At least one of \dQuote{\code{counts}} or \dQuote{\code{data}} is required
+    #' @param cells_index Name of column in \code{obs} to add as cell names
+    #' @param features_index Name of column in \code{var} to add as feature names
+    #' @param obs_column_names Names of columns in \code{obs} to add as
+    #' cell-level meta data
+    #' @param var_column_names Names of columns in \code{var} to add as
+    #' feature-level meta data
+    #' @param embeddings Names of arrays in \code{obsm} to load in as the
+    #' cell embeddings; pass \code{FALSE} to suppress loading in any
+    #' dimensional reductions
+    #' @param loadings Named vector of arrays in \code{varm} to load in as the
+    #' feature loadings; names must be names of array in \code{obsm} (eg.
+    #' \code{loadings = c(X_pca = 'PCs')}); will try to determine
+    #' \code{loadings} from \code{embeddings}
+    #' @param graphs Names of arrays in \code{obsp} to load in as
+    #' \code{\link[SeuratObject]{Graph}s}
     #'
-    #' @return ...
+    #' @return A \code{\link[SeuratObject]{Seurat}} object
     #'
-    to_seurat = function(X_layers = c(counts = 'counts', data = 'logcounts')) {
+    to_seurat = function(
+      X_layers = c(counts = 'counts', data = 'logcounts'),
+      cells_index = NULL,
+      features_index = NULL,
+      obs_column_names = NULL,
+      var_column_names = NULL,
+      embeddings = NULL,
+      loadings = NULL,
+      graphs = NULL
+    ) {
       .check_seurat_installed()
-      .NotYetImplemented()
+      stopifnot(
+        "'cells_index' must be a single character value" = is.null(cells_index) ||
+          (is_scalar_character(cells_index) && !is.na(cells_index)),
+        "'obs_column_names' must be a character vector" = is.null(obs_column_names) ||
+          is.character(obs_column_names) ||
+          is_scalar_logical(obs_column_names),
+        "'embeddings' must be a character vector" = is.null(embeddings) ||
+          is.character(embeddings) ||
+          is_scalar_logical(embeddings),
+        "'loadings' must be a named character vector" = is.null(loadings) ||
+          (is.character(loadings) && is_named(loadings, allow_empty = FALSE)) ||
+          is_scalar_logical(loadings),
+        "'graphs' must be a character vector" = is.null(graphs) ||
+          is.character(graphs) ||
+          is_scalar_logical(graphs)
+      )
+      tryCatch(self$obs_df, error = \(...) stop("No 'obs' found", call. = FALSE))
```

Comment:
Writing an anonymous function this way is nice but creates a hard depends on R 4.1.0.  _That is probably fine_ but if we wanted to be more defensive we could the keyword `function` in.